content/src/content/docs/docs/ai/getting-started.mdx

```diff
@@ -0,0 +1,466 @@
+
+<Aside type="note" title="Declarative LLM Interactions">
+  Notice that the above code does not know or care which LLM provider (OpenAi, Anthropic, etc.) will be used. Instead, we focus on _what_ we want to accomplish (i.e. our business logic), not _how_ to accomplish it.
+</Aside>
+
+## Select an LLM Model
+
+Next, we need to select which provider we want to use to satisfy the `Completions` requirement of our `generateDadJoke` program:
+
+```ts twoslash {1,17,25,27} collapse={5-10}
+import { OpenAiCompletions } from "@effect/ai-openai"
+import { Completions } from "@effect/ai"
+import { Console, Effect } from "effect"
+
+const generateDadJoke = Effect.gen(function*() {
+  const completions = yield* Completions.Completions
+  return yield* completions.create("Generate a hilarious dad joke").pipe(
+    Effect.andThen((response) => Console.log(response.text))
+  )
+})
+
+// Create an `AiModel` which provides a concrete implementation of 
+// `Completions` and requires an `OpenAiClient`
+//
+//      ┌─── AiModel<Completions, OpenAiClient>
+//      ▼
+const Gpt4o = OpenAiCompletions.model("gpt-4o")
+
+// Building an `AiModel` adds its requirements to the program
+//
+//     ┌─── Effect<void, AiError, OpenAiClient>
+//     ▼
+const main = Effect.gen(function*() {
+  // Build the `AiModel` into a `Provider`
+  const gpt4o = yield* Gpt4o
+  // Provide the implementation of `Completions` to `generateDadJoke`
+  yield* gpt4o.provide(generateDadJoke)
+})
+```
+
+Before we can finish satisfying requirements and make our program runnable, it is important that we understand the purpose of the `AiModel` data type.
+
+## Understanding `AiModel`
+
+The `AiModel` data type represents a **provider-specific implementation** of one or more services, such as `Completions` or `Embeddings`. It is the primary way that you can plug a real LLM (like OpenAI’s GPT-4o) into your program.
+
+```ts showLineNumbers=false
+export interface AiModel<Provides, Requires> {}
+```
+
+An `AiModel` has two generic type parameters:
+
+- **Provides** - the services this model will provide when built
+- **Requires** - the services this model will require to be built
+
+This allows Effect to track which services should be added to the requirements of the program that builds the `AiModel`, as well as which services the built `AiModel` can provide.
+
+### Creating an `AiModel`
+
+To create an `AiModel`, you can use the model-specific factory from one of Effect's provider integration packages. 
+
+For example:
+
+```ts showLineNumbers=false
+import { OpenAiCompletions } from "@effect/ai-openai"
+
+//      ┌─── AiModel<Completions, OpenAiClient>
+//      ▼
+const Gpt4o = OpenAiCompletions.model("gpt-4o")
+```
+
+This creates an `AiModel` that:
+
+- **Provides** an OpenAi-specific implementation of the `Completions` service using `"gpt-4o"`
+- **Requires** an `OpenAiClient` to be built
+
+### Building an `AiModel`
+
+When you build an `AiModel`, you get back a `Provider<Provides>`:
```

Comment:
"build" sounds a bit off to me here, I'm just yielding, but "build" makes it sound like I'm doing something more involved

Reply:
The terminology is somewhat unfortunate I suppose, but that is precisely what we are doing - we're building the `AiModel` into a `Provider` which can provide the requisite services.